python/cudf/cudf/core/join/join.py

```diff
@@ -0,0 +1,431 @@
+                if name in left_on and name in right_on:
+                    if left_on.index(name) == right_on.index(name):
+                        no_suffix_cols.append(name)
+
+        for name in same_named_columns:
+            if name not in no_suffix_cols:
+                self.lhs.rename({name: "%s%s" % (name, lsuffix)}, inplace=True)
+                self.rhs.rename({name: "%s%s" % (name, rsuffix)}, inplace=True)
+                if name in left_on:
+                    left_on[left_on.index(name)] = "%s%s" % (name, lsuffix)
+                if name in right_on:
+                    right_on[right_on.index(name)] = "%s%s" % (name, rsuffix)
+
+        self.left_on = left_on if left_on is not None else []
+        self.right_on = right_on if left_on is not None else []
+        self.lsuffix = lsuffix
+        self.rsuffix = rsuffix
+
+    @staticmethod
+    def validate_merge_cfg(
+        lhs,
+        rhs,
+        on,
+        left_on,
+        right_on,
+        left_index,
+        right_index,
+        how,
+        lsuffix,
+        rsuffix,
+    ):
+        """
+        Error for various combinations of merge input parameters
+        """
+
+        # must actually support the requested merge type
+        if how not in ["left", "inner", "outer", "leftanti", "leftsemi"]:
+            raise NotImplementedError(
+                "{!r} merge not supported yet".format(how)
+            )
+
+        # Passing 'on' with 'left_on' or 'right_on' is ambiguous
+        if on is not None:
+            if left_on is not None or right_on is not None:
+                raise ValueError(
+                    'Can only pass argument "on" OR "left_on" '
+                    'and "right_on", not a combination of both.'
+                )
+            # Keys in 'on' need to be in both operands
+            on_keys = [on] if not isinstance(on, list) else on
+            for key in on_keys:
+                if not (key in lhs._data.keys() and key in rhs._data.keys()):
+                    raise KeyError("on key {} not in both operands".format(on))
+
+        # in this case, keys must be present in their corresponding operands
+        if left_on is not None and right_on is not None:
+            left_on_keys = (
+                [left_on] if not isinstance(left_on, list) else left_on
+            )
+            right_on_keys = (
+                [right_on] if not isinstance(right_on, list) else right_on
+            )
+
+            for key in left_on_keys:
+                if key not in lhs._data.keys():
+                    raise KeyError('Key "{}" not in left operand'.format(key))
+            for key in right_on_keys:
+                if key not in rhs._data.keys():
+                    raise KeyError('Key "{}" not in right operand'.format(key))
+
+        # Require same total number of columns to join on in both operands
+        if left_on is None:
+            len_left_on = 0
+        elif not isinstance(left_on, list):
```

Comment:
What if `left_on` is a tuple or other list-like? May want to use `pd.api.types.is_list_like`

Reply:
Updated